### PR TITLE
ci: setup AppVeyor

### PR DESCRIPTION
**Pull request by kalikiana**
- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] If this is a bugfix. Have you checked that there is a bug report open for the issue you are trying to fix on [bug reports](https://bugs.launchpad.net/snapcraft)?
- [x] If this is a new feature. Have you discussed the design on the [forum](https://forum.snapcraft.io)?
- [x] Have you successfully run `./runtests.sh static`?
- [x] Have you successfully run `./runtests.sh unit`?

-----
This PR adds a configuration file for CI in AppVeyor. To support integration tests on non-unix platforms I'm splitting out the fixtures that require unix sockets and moving pexpect away from unix-specifc API.

Requires: #2096

No tests were added or removed. However the store login method used for testing uses `SNAPCRAFT_TEST_INPUT` now to force regular input for passwords to avoid depending on a `tty` which is platform-specific.

I locally ran the tests:
 - `./runtests.sh tests/unit` with unrelated failures in [tests.unit.test_lifecycle.CoreSetupTestCase.test_core_setup_if_docker_env](https://bugs.launchpad.net/snapcraft/+bug/1752576), `tests.unit.test_mangling.TestClearExecstack.test_execstack_clears` and `tests.unit.test_elf`.
 - `./runtests.sh tests/integration` with one unrelated failure in [tests.integration.general.test_parser.TestParserWikis](https://bugs.launchpad.net/snapcraft/+bug/1752580)
 - `./runtests.sh static`: Everything passed

### CODE REVIEW

**Comment by sergiusens** [comment]
to make the change easier just create the package and add all the classes in `__init__.py` using the logic described for `FakeSnapd` or make use of `from _fixtures import *`

File: tests/fixture_setup/__init__.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import socket
+
+if hasattr(socket, "AF_UNIX"):
+    from ._unix import FakeSnapd  # noqa
+from ._fixtures import (  # noqa
```
Comment (sergiusens) [inline tests/fixture_setup/__init__.py]: add the specific code for the ignored error

File: tests/fixture_setup/__init__.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import socket
+
+if hasattr(socket, "AF_UNIX"):
+    from ._unix import FakeSnapd  # noqa
```
Comment (sergiusens) [inline tests/fixture_setup/__init__.py]: add the specific code for the ignored error

File: tests/integration/__init__.py [hunk 3]
```diff
@@ -39,6 +41,7 @@
 from testtools.matchers import MatchesRegex
 
 from snapcraft.internal.os_release import OsRelease
+from snapcraft.internal import errors
```
Comment (sergiusens) [inline tests/integration/__init__.py]: this is wrong, we should not import snapcraft into the integration tests.

Comment (kalikiana) [reply tests/integration/__init__.py]: I fixed this in #2096 and re-based this one on top of it.

File: appveyor.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+build: off
+before_test:
+- cmd: |
+    %PYTHON%\\python.exe -m venv venv
+    venv\\Scripts\\activate.bat
+    ECHO %VIRTUAL_ENV%
+    python -c "import sys; print(sys.executable)"
+    python -m pip install -r requirements.txt
+    python -m pip install -r requirements-devel.txt
+    python setup.py install
```
Comment (sergiusens) [inline appveyor.yml]: this might be better as `python -m pip install .` or does that not work on windows?

Comment (kalikiana) [reply appveyor.yml]: It didn't seem to work when I first tried it. Apparently with `--prefix` it does. Unfortunately we're not getting an `.exe` this way so I need to execute Snapcraft in the tests via `python -m` now.

**Pull request merged**